geo-location.html

```diff
@@ -142,16 +142,30 @@
        *   @param {Number} detail.longitude Longitude of the current position.
        */
 
+      detached: function () {
+        if (watch_ !== null) {
+          this._clearWatch(watch_);
+        }
+      },
+
+      /**
+       * Stop updating latitude/longitude as the device changes position.
+       * @param {Number} watch watch ID value.
+       */
+      _clearWatch: function (watch) {
+        navigator.geolocation.clearWatch(watch);
+        watch_ = null;
+      },
+
       clear: function () {
         this._setPosition(null);
         this._setLatitude(null);
         this._setLongitude(null);
       },
 
       fetch: function () {
-        if (watch_) {
-          navigator.geolocation.clearWatch(watch_);
-          watch_ = null;
+        if (watch_ !== null) {
```

Comment:
How about this? Do we actually need to clear existing watch everytime we run `fetch`?